frontend_tests/node_tests/drafts.js

```diff
@@ -443,3 +453,145 @@ test("format_drafts", ({override_rewire, mock_template}) => {
+        updatedAt: date(-10),
+    };
+    const pm_draft_2 = {
+        private_message_recipient: "aaron@zulip.com",
+        reply_to: "iago@zulip.com",
+        type: "private",
+        content: "Test private message 2",
+        updatedAt: date(-5),
+    };
+    const pm_draft_3 = {
+        private_message_recipient: "aaron@zulip.com",
+        reply_to: "zoe@zulip.com",
+        type: "private",
+        content: "Test private message 3",
+        updatedAt: date(-2),
+    };
+
+    const expected_pm_drafts = [
+        {
+            draft_id: "id2",
+            is_stream: false,
+            recipients: "Aaron",
+            raw_content: "Test private message",
+            time_stamp: "Jan 30",
+        },
+        {
+            draft_id: "id5",
+            is_stream: false,
+            recipients: "Aaron",
+            raw_content: "Test private message 3",
+            time_stamp: "Jan 29",
+        },
+        {
+            draft_id: "id4",
+            is_stream: false,
+            recipients: "Aaron",
+            raw_content: "Test private message 2",
+            time_stamp: "Jan 26",
+        },
+    ];
+
+    const expected_other_drafts = [
+        {
+            draft_id: "id1",
+            is_stream: true,
+            stream_name: "stream",
+            stream_color: "#FFFFFF",
+            dark_background: "",
+            topic: "topic",
+            raw_content: "Test stream message",
+            time_stamp: "7:55 AM",
+        },
+        {
+            draft_id: "id3",
+            is_stream: true,
+            stream_name: "stream 2",
+            stream_color: "#FFFFFF",
+            dark_background: "",
+            topic: "topic",
+            raw_content: "Test stream message 2",
+            time_stamp: "Jan 21",
+        },
+    ];
+
+    $("#drafts_table").append = noop;
+
+    const draft_model = drafts.draft_model;
+    const ls = localstorage();
+    const data = {
+        id1: stream_draft_1,
+        id2: pm_draft_1,
+        id3: stream_draft_2,
+        id4: pm_draft_2,
+        id5: pm_draft_3,
+    };
+    ls.set("drafts", data);
+    assert.deepEqual(draft_model.get(), data);
+
+    const stub_render_now = timerender.render_now;
+    override_rewire(timerender, "render_now", (time) =>
+        stub_render_now(time, new Date(1549958107000)),
+    );
+
+    sub_store.get = function (stream_id) {
+        assert.equal(stream_id, 30);
+        return {name: "stream"};
+    };
+
+    mock_template("draft_table_body.hbs", false, (data) => {
+        // Tests splitting up drafts by current narrow.
+        assert.deepEqual(data.narrow_drafts, expected_pm_drafts);
+        assert.deepEqual(data.other_drafts, expected_other_drafts);
+        return "<draft table stub>";
+    });
+
+    override_rewire(drafts, "set_initial_element", noop);
+
+    override(compose_state, "composing", () => true);
+    override(compose_state, "get_message_type", () => "private");
```

Comment:
I see we use this pattern already in this test file, but probably using the real `compose_state` module with just calling `compose_state.set_message_type("private")` would let us remove these overrides.